pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,73 @@
 Contains the MPSPrep template.
 """
 
+import numpy as np
+
 import pennylane as qml
 from pennylane.operation import Operation
 from pennylane.wires import Wires
 
 
+def _complete_unitary(columns):
+    """
+    Completes a unitary matrix given a list of orthonormal columns.
+
+    Args:
+        columns (List[Array]): List of initial orthonormal columns of dimension d.
+
+    Returns:
+        Array: Completed unitary matrix of dimension :math:`(d, d)`.
+    """
+
+    columns = qml.math.stack(columns).T
+    d = columns.shape[0]
+    k = columns.shape[1]
+
+    unitary = qml.math.zeros_like(columns @ columns.T)
+
+    if qml.math.get_interface(columns) == "jax":
+        unitary = unitary.at[:, :k].set(columns)
+
+    else:
+        unitary[:, :k] = columns
+
+    # Complete the remaining columns using Gram-Schmidt
+    rng = np.random.default_rng(42)
+    new_columns = qml.math.array(rng.random((d, d - k)))
+
+    ortogonal_projection_matrix = qml.math.eye(d) - qml.math.dot(unitary, qml.math.conj(unitary.T))
+    new_columns = qml.math.dot(ortogonal_projection_matrix, new_columns)
+
+    q, _ = qml.math.linalg.qr(
+        new_columns
+    )  # apply QR decomposition to orthonormalize the new columns
+
+    if qml.math.get_interface(unitary) == "jax":
+        unitary = unitary.at[:, k:d].set(q[:, : d - k])
+    else:
+        unitary[:, k:d] = q[:, : d - k]
+    return unitary
+
+
 class MPSPrep(Operation):
     r"""Prepares an initial state from a matrix product state (MPS) representation.
 
     .. note::
 
-        Currently, this operator can only be used with ``qml.device(“lightning.tensor”)``.
+        Tensor simulators are designed to run MPS structures more efficiently.
+        However, it may be useful to use state vector simulator if you are looking for a gate decomposition that
+        prepares the mps in a quantum circuit. For the gate decomposition is required to introduce work qubits
+        for the decomposition.
```

Comment:
Might want to paraphrase this to say that for non-lightning.tensor devices, this operation uses a gate decomposition that requires work (auxiliary) qubits to be specified.